src/tribler/core/components/libtorrent/torrentdef.py

```diff
@@ -184,12 +203,8 @@ def get_trackers_as_single_tuple(self):
         Returns a flat tuple of all known trackers.
         """
         if self.get_tracker_hierarchy():
-            trackers = []
-            for level in self.get_tracker_hierarchy():
-                for tracker in level:
-                    if tracker and tracker not in trackers:
-                        trackers.append(tracker)
-            return tuple(trackers)
+            trackers = itertools.chain.from_iterable(self.get_tracker_hierarchy())
+            return tuple(set(filter(None, trackers)))
```

Comment:
This is a nice simplification!
This code is correct assuming that the order of the tuple isn't important, as the returned value now doesn't maintain the same tracker order as presented in `self.get_tracker_hierarchy()`.

However, this assumption is valid as all places that use `get_trackers_as_single_tuple()` convert it to a set anyway.

To prove the correctness of the code, please consider adding a test similar to the one presented below, which demonstrates that both versions of the code (original and new) work as expected:

```python
def test_get_trackers_as_single_tuple(tdef):
    # Test that `get_trackers_as_single_tuple` returns flat tuple of trackers
    tdef.get_tracker_hierarchy = Mock(return_value=[['t1', 't2'], ['t3'], ['t4']])
    t = tdef.get_trackers_as_single_tuple()
    assert set(t) == {'t1', 't2', 't3', 't4'}
```

If you'd like, you could further refine the refactoring process by reworking the entirety of the `get_trackers_as_single_tuple` function. It's now clear that the return type can be changed from a `tuple` to a `set`.

Reply:
Thank you! 😄 I'll add such a test. I'll check where I can change the return type.

Reply:
You are absolutely correct :)